Dockerfile

```diff
@@ -1,11 +1,35 @@
-FROM node:8-slim
-ENV NPM_CONFIG_LOGLEVEL error
+#
+# --- Base Node Image ---
+FROM node:8-alpine AS base
+
 WORKDIR /src
-ADD . /src
-RUN cd /src \
- && npm install \
- && npm run build \
- && npm cache clear --force \
- && rm -rf ~/.npm \
- && rm -rf /var/lib/apt/lists/*
-ENTRYPOINT ["npm", "run", "dashboard"]
+
+# Copy package.json first to benefit from layer caching
+COPY package*.json ./
+RUN npm install --only=production
+# Copy production node_modules aside for later
+RUN cp -R node_modules prod_node_modules
+# Install remaining dev dependencies
+RUN npm install
+
+COPY . /src
+
+# Run all webpack build steps
+RUN npm run prepublish && npm run build
+
+
+#
+# --- Production Image ---
+FROM node:8-alpine AS release
+WORKDIR /src
+
+# Copy production node_modules
+COPY --from=base /src/prod_node_modules /src/node_modules
+COPY --from=base /src/package*.json /src/
+
+# Copy compiled src dirs
+COPY --from=base /src/Parse-Dashboard/ /src/Parse-Dashboard/
+
+
+EXPOSE 4040
```

Comment:
4040 is only the default port. Is it relevant if the users has a config that listens to another port?

Reply:
Yep, good spot!  I don't think there is much point a user changing the listening port when running in Docker seeing as they can map the container port to a port of their choice on the host system, but to support this use case (and people porting their existing config to Docker) I've removed the default 4040 from the Dockerfile